src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCallerModelState.java

```diff
@@ -0,0 +1,148 @@
+package org.broadinstitute.hellbender.tools.exome.cnlohcaller;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.broadinstitute.hellbender.tools.exome.ACNVModeledSegment;
+import org.broadinstitute.hellbender.utils.GATKProtectedMathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.IntStream;
+
+
+public class CNLOHCallerModelState implements Serializable {
+    private CNLOHCallerModelState() {}
+
+    static final long serialVersionUID = 33733733712L;
+
+    final static private int MIN_CN = 0;
+    final static private int MAX_CN = 5;
+
+    private double eAlpha;
+
+    /** Number of rho values we will use.  This implicitly dictates the maximum number of unique values we expect rho
+     * to be able to take.
+     */
+    private int numK;
+
+    /** Current unique values of rho in this sample.  For each segment, rho is CCF*purity.  Note that each
+     * segment will use a rho from this list.
+     */
+    private double[] rhos;
+
+    /** likelihood of CN 0... for all segments.  In other words, all segments share the pis. The keys are the actual
+     * copy numbers we are allowing.  The values are the pi (likelihood of that copy number).
+     *
+     * A SortedMap is best for this map.
+     */
+    private SortedMap<Integer, Double> cnToPiMap = new TreeMap<>();
+
+    /** ploidy * CCF * purity
+     *
+     * Please note that this does not change in the model.  This is a placeholder.
+     */
+    private double lambda;
+
+    /** Log'ed likelihood of each rho.  Will always be of length K */
+    private double[] ELnPhiK;
```

Comment:
Actually, I like the terminology you use in `CNLOHCaller.java` where you call phi and pi "discrete pdfs".  I like to reserve likelihood for something involving the observed data.  So the javadoc here would be effective discrete pdf on rhos.